exporter/awsemfexporter/metric_translator.go

```diff
@@ -490,10 +493,13 @@ func translateCWMetricToEMF(cWMetric *cWMetrics, config *Config) (*cwlogs.Event,
 func translateGroupedMetricToEmf(groupedMetric *groupedMetric, config *Config, defaultLogStream string) (*cwlogs.Event, error) {
 	cWMetric := translateGroupedMetricToCWMetric(groupedMetric, config)
 	event, err := translateCWMetricToEMF(cWMetric, config)
-
 	if err != nil {
 		return nil, err
 	}
+	// Drop a nil putLogEvent for EnhancedContainerInsights
+	if config.EnhancedContainerInsights && event == nil {
+		return nil, errMissingMetricsForEnhancedContainerInsights
```

Comment:
nit: This could be returned from the `translateCWMetricToEMF` instead of `nil, nil`.